src/radical/pilot/agent/lm/spark.py

```diff
@@ -0,0 +1,285 @@
+                if not os.path.isfile('scala-2.10.4.tgz'):
+                    raise RuntimeError("Scala wasn't downloaded properly. Please try again")
+                os.system('tar -xvf scala-2.10.4.tgz ; cd scala-2.10.4 ; export PATH=`pwd`/bin:$PATH; export SCALA_HOME=`pwd`')
+                os.system('rm scala-2.10.4.tgz')
+                scala_home = os.getcwd() + '/scala-2.10.4'
+                os.system('cd')
+
+
+            if lrms.node_list[0]!='localhost':
+                hostname = subprocess.check_output('/bin/hostname').split(lrms.node_list[0])[1].split('\n')[0]
+
+            else:
+                hostname = ''
+
+            spark_conf_slaves = open(spark_home+"/conf/slaves",'w')
+
+            if len(lrms.node_list) == 1:
+                spark_conf_slaves.write(lrms.node_list[0]+hostname)
+                spark_conf_slaves.write('\n')
+            else:
+                for nodename in lrms.node_list[1:]:
+                    spark_conf_slaves.write(nodename+hostname)
+                    spark_conf_slaves.write('\n')
+
+            spark_conf_slaves.close()
+
+            ## put Master Ip in spark-env.sh file - 
+
+            python_path = os.getenv('PYTHONPATH')
+            python = ru.which('python')
+            logger.info('Python Executable: %s' % python)
+            master_ip = lrms.node_list[0]+hostname
+
+            #Setup default env properties:
+            spark_default_file = open(spark_home + "/conf/spark-defaults.conf",'w')
+            spark_master_string = 'spark://%s:7077' % master_ip
+            spark_default_file.write('spark.master  ' + spark_master_string + '\n')
+            spark_default_file.close()
+            logger.info("Let's print the config")
+            logger.info('Config : {0}'.format(cfg['resource_cfg']))
+
+            spark_env_file = open(spark_home + "/conf/spark-env.sh",'w')
+            #load in the spark enviroment of master and slaves the
+            #configurations of the machine
+            if master_ip!='localhost':
+                for config in cfg['resource_cfg']['pre_bootstrap_1']:
+                    spark_env_file.write(config + '\n')
+
+            spark_env_file.write('export SPARK_MASTER_IP=' + master_ip + "\n")
+            spark_env_file.write('export SCALA_HOME='+ scala_home+ "\n")
+            spark_env_file.write('export JAVA_HOME=' + java_home + "\n")
+            spark_env_file.write('export SPARK_LOG_DIR='+os.getcwd()+'/spark-logs'+'\n')
+            spark_env_file.write('export PYSPARK_PYTHON='+python+'\n')
+
+            spark_env_file.close()
+
+
+            #### Start spark Cluster
+            spark_start = subprocess.check_output(spark_home + '/sbin/start-all.sh')
+            if 'Error' in spark_start:
+                raise RuntimeError("Spark Cluster failed to start: %s" % spark_start)
+            else:
+                logger.info('Start Spark Cluster')
+            launch_command = spark_home +'/bin'
```

Comment:
I see - but then its not a `launch_command`, but a `launch_path`?  Sorry for being picky, but it would be good if we can still understand the code a couple of months from now... :)